From 79b306fc51b037b186e46dc6643e7ece4a75c958 Mon Sep 17 00:00:00 2001
From: Tomasz Figa <t.figa@samsung.com>
Date: Fri, 1 Mar 2013 15:27:05 +0100
Subject: [PATCH 0152/1302] gpu: drm: exynos: fimd: Parse display physical size
 from DT

This patch adds a temporary solution for getting physical display size
from device tree by parsing driver-specific properties in display node.

Signed-off-by: Tomasz Figa <t.figa@samsung.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/gpu/drm/exynos/exynos_drm_fimd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/exynos/exynos_drm_fimd.c b/drivers/gpu/drm/exynos/exynos_drm_fimd.c
index 0d7e42b..e8465f5 100644
--- a/drivers/gpu/drm/exynos/exynos_drm_fimd.c
+++ b/drivers/gpu/drm/exynos/exynos_drm_fimd.c
@@ -932,6 +932,12 @@ static int fimd_probe(struct platform_device *pdev)
 			DRM_ERROR("failed: of_get_fb_videomode() : %d\n", ret);
 			return ret;
 		}
+
+		/* FIXME */
+		of_property_read_u32(display_np, "samsung,panel-width-mm",
+						&pdata->panel.width_mm);
+		of_property_read_u32(display_np, "samsung,panel-height-mm",
+						&pdata->panel.height_mm);
 	} else {
 		pdata = dev->platform_data;
 		if (!pdata) {
-- 
1.8.3.2

